ocs_ci/ocs/utils.py

```diff
@@ -1266,3 +1271,39 @@ def get_primary_cluster_config():
     for cluster in ocsci_config.clusters:
         if cluster.MULTICLUSTER["primary_cluster"]:
             return cluster
+
+
+def collect_pod_container_rpm_package(dir_name):
+    """
+    Collect information about rpm packages from all containers from
+
+    Args:
+        dir_name(str): directory to store container rpm package info
+
+    """
+    from ocs_ci.ocs.resources import pod
+
+    create_directory_path(dir_name)
+    log.info(f"Directory path for rpm logs is{dir_name}")
+    pods = pod.get_all_pods(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+    for pod_name in pods:
+        pod_object = pod_name.get()
+        pod_containers = pod_object.get("spec").get("containers")
+        for container in pod_containers:
+            container_name = container["name"]
+            ocp_pod_obj = OCP(
+                kind=constants.POD, namespace=defaults.ROOK_CLUSTER_NAMESPACE
+            )
+            pod_status = ocp_pod_obj.get_resource_status(pod_name.name)
+            if pod_status == constants.STATUS_RUNNING:
+                command = f"exec -i {pod_name.name} -c {container_name} -- rpm -qa"
+                ocp_obj = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+                container_output = ocp_obj.exec_oc_cmd(command)
+                log_file_name = f"{dir_name}/{container_name}-rpm.log"
+                try:
+                    with open(log_file_name, "w") as f:
+                        f.write(container_output)
+                        f.close()
+                except Exception:
+                    log.warning(f"Cannot write the log to the file {log_file_name}")
+    log.info("Retrived rpm package list from all containers successfully")
```

Comment:
If you will move the try logic inside to this function you cannot say that here as some could fail.

Even now if you collected the logs and didn't write to the file it will write that it successfully retrieved list but didn't write it as it failed and you just logged warning.

I guess this is not typical that you should handle such exception here, or you are failing to write a log file from some rason?

If so, this should be probably covered in general wrap of try except block on the place where you will call that to not break the session scope fixture.

Reply:
Updated with suggested changes